### ISSUE TITLE
#2085 breaks iOS build of darwinssl.c

### ISSUE THREAD

**Issue by chris-araman** [curl/curl#2396]
### I did this
Updated from curl 7.58.0 to 7.59.0. Tried building for iOS:
```
    ./configure --disable-shared --enable-static ${HOST} --enable-ares="${SDK_ROOT}" --with-darwinssl \
    --enable-ipv6 --disable-dict --disable-file --disable-ftp --disable-gopher --disable-imap --disable-ldap \
    --disable-ldaps --disable-pop3 --disable-rtsp --disable-smb --disable-smtp --disable-telnet --disable-tftp \
    --enable-threaded-resolver=no \
    --prefix="${SDK_ROOT}" \
    --without-libidn2
    make
```

Build failed with:
```
vtls/darwinssl.c:1137:6: warning: 'CFURLCreateDataAndPropertiesFromResource' is deprecated: first deprecated in iOS 7.0 - For resource data, use the CFReadStream API. For file resource properties, use CFURLCopyResourcePropertiesForKeys. [-Wdeprecated-declarations]
  if(CFURLCreateDataAndPropertiesFromResource(NULL, pkcs_url, &pkcs_data,
     ^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFURLAccess.h:52:9: note: 'CFURLCreateDataAndPropertiesFromResource' has been explicitly marked deprecated here
Boolean CFURLCreateDataAndPropertiesFromResource(CFAllocatorRef alloc, CFURLRef url, CFDataRef *resourceData, CFDictionaryRef *properties, CFArrayRef desiredProperties, SInt32 *errorCode) API_DEPRECATED("For resource data, use the CFReadStream API. For file resource properties, use CFURLCopyResourcePropertiesForKeys.", macos(10.0,10.9), ios(2.0,7.0), watchos(2.0,2.0), tvos(9.0,9.0));
        ^
vtls/darwinssl.c:1199:20: error: implicit declaration of function 'SecIdentityCreateWithCertificate' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
          status = SecIdentityCreateWithCertificate(NULL,
                   ^
vtls/darwinssl.c:1199:20: note: did you mean 'SecTrustCreateWithCertificates'?
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk/System/Library/Frameworks/Security.framework/Headers/SecTrust.h:206:10: note: 'SecTrustCreateWithCertificates' declared here
OSStatus SecTrustCreateWithCertificates(CFTypeRef certificates,
```

### I expected the following
Successful build.

### curl/libcurl version
7.59.0

### operating system
build host: macOS 10.13.3 with Xcode 9.2
iOS target: `-mios-version-min=10.0`

It seems that PR #2085 introduced the call to `SecIdentityCreateWithCertificate` for both iOS and macOS, but it's only available on macOS 10.5+ per Apple's documentation:
https://developer.apple.com/documentation/security/1401160-secidentitycreatewithcertificate?language=objc

@refnum, is there an alternative API that can be used for iOS?

**Comment by chris-araman** [comment]
I think the fix here is just to include the `SecCertificateGetTypeID` block only in Mac build that corresponds to `SecItemImport` above, as `SecPKCS12Import` is documented as returning `CFDictionaryGetTypeID` items. @refnum, does that sound right to you?

**Comment by bagder** [comment]
Fixed in b7b2809a212a6 via #2397
